wade/chain.py

```diff
@@ -157,8 +163,12 @@ def state_prepare(self):
         directly to commit.
 
         """
+        # If a sync command traverses a chain it disables updates for that chain
+        # on the way in.
+        if self._op_type == SYNC_OP:
+            self._accept_updates_by_chain[self._cmd.obj_id] = False
```

Comment:
Is there a risk in disabling updates on "prepare"? I don't think this counts as a regular state changing action so I think it's safe. This way while the sync command is in flight updates are disabled.